File: Website/js/Packets/readLine.js
ReadLinePacket = (function()
{
	function ReadLinePacket()
	{

	}

	ReadLinePacket.prototype.handle = function(client, message)
	{
		webconsole.readLine(function(line)
		{
			client.send("readLine",
				{
					"line": line
				});
		}, message.intercept);
	};

	return ReadLinePacket;
})();

File: ServerCSharp/Packets/ReadLine.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ServerCSharp.Packets
{
    public class ReadLine : Packet
    {
        public ReadLine(Server server) : base(server, "line") { }

        public override void Handle(Client client, Message message)
        {
            if (client.PendingReadLine != null)
            {
                var callback = client.PendingReadLine;
                client.PendingReadLine = null;
                callback(message["line"] as string);
            }
        }
    }
}